### ISSUE TITLE
Link URL getting encoded in react-markdown v9

### ISSUE THREAD

**Issue by ankurparihar** [remarkjs/react-markdown#797]
### Initial checklist

- [X] I read the [support docs](https://github.com/remarkjs/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/remarkjs/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/remarkjs/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Aremarkjs&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

9.0.0

### Steps to reproduce

Go to: https://remarkjs.github.io/react-markdown/

Open Network panel in DevTools

Paste this in readme

```
[![Test Badge](
https://github.com/remarkjs/react-markdown/workflows/main/badge.svg?v=123&branch=feat/router)](https://github.com/remarkjs/react-markdown/actions)
```

Observed URL: https://github.com/remarkjs/react-markdown/workflows/main/badge.svg?v=123&amp;branch=feat/router

Notice: `&amp;`, this is causing issues

### Expected behavior

URL should not be double encoded

### Actual behavior

URL is being encoded in a weird way.

**Comment by wooorm** [comment]
> this is causing issues

It is not.

This is how HTML works.

**Comment by ankurparihar** [comment]
Hey, This is causing issues due to invalid URL.

Here's a screenshot for the comparison on your own site, please have a look.

<img width="1728" alt="Screenshot 2023-11-13 at 2 23 25 PM" src="https://github.com/remarkjs/react-markdown/assets/26161041/a9e3d760-ab64-442c-9de0-0b1b5118158e">


Can you please help me with a correct URL, with parameters.

**Comment by ankurparihar** [comment]
Never mind, I'll recreate another example. But this is breaking!

**Comment by wooorm** [comment]
It is not

**Comment by ankurparihar** [comment]
Ok, finally, I'm able to reproduce!

Here you can see, in the markdown I've added a normal [url](https://sonarqube-internal.phonepe.com/api/project_badges/measure?branch=test/sonar-badges&project=phonepe:web-starter&metric=coverage&token=sqb_6258bc87f5021a0480d4e2e5ea59d37ce4af6f78) taken from official [sonarqube project](https://next.sonarqube.com/sonarqube/project/information?id=sonarqube)

![Screenshot 2023-11-13 at 3 58 10 PM](https://github.com/remarkjs/react-markdown/assets/26161041/9a642fdf-b4c2-4bda-be47-497d03d7572e)

You can see what's going on in network tab, the & gets converted to &amp; and in-turn becomes an invalid query parameter.

I'm willing to contribute a fix, but I have no idea where it's getting encoded from like this.

**Comment by wooorm** [comment]
Paste code so I can try it out

**Comment by wooorm** [comment]
Ohh, managed to reproduce!

**Comment by ankurparihar** [comment]
Sorry, I was out for a while,

Here's a snippet you can try - 

`[![Coverage](https://next.sonarqube.com/sonarqube/api/project_badges/measure?project=sonarqube&metric=coverage&token=d95182127dd5583f57578d769b511660601a8547)](https://next.sonarqube.com/sonarqube/dashboard?id=sonarqube)
`

**Comment by wooorm** [comment]
Released in [9.0.1](https://github.com/remarkjs/react-markdown/releases/tag/9.0.1)!

**Comment by ankurparihar** [comment]
Awesome, thank you so much @wooorm 🙌
